File: demo-vault/code/hello.java
import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

/**
 * A sample Java class demonstrating OOP patterns and modern Java features.
 */
public class TaskManager {

    public enum Priority { LOW, MEDIUM, HIGH }

    public record Task(String title, Priority priority, boolean done) {
        public Task markDone() {
            return new Task(title, priority, true);
        }
        public String summary() {
            return String.format("[%s] %s (%s)", done ? "x" : " ", title, priority);
        }
    }

    private final List<Task> tasks;

    public TaskManager(List<Task> tasks) {
        this.tasks = tasks;
    }

    public List<Task> filterByPriority(Priority priority) {
        return tasks.stream()
            .filter(t -> t.priority() == priority)
            .collect(Collectors.toList());
    }

    public List<Task> pendingTasks() {
        return tasks.stream()
            .filter(t -> !t.done())
            .collect(Collectors.toList());
    }

    public Optional<Task> findTask(String title) {
        return tasks.stream()
            .filter(t -> t.title().equalsIgnoreCase(title))
            .findFirst();
    }

    public static void main(String[] args) {
        var manager = new TaskManager(List.of(
            new Task("Write tests", Priority.HIGH, false),
            new Task("Fix login bug", Priority.HIGH, true),
            new Task("Update docs", Priority.LOW, false)
        ));

        System.out.println("Pending:");
        manager.pendingTasks().forEach(t -> System.out.println("  " + t.summary()));

        var found = manager.findTask("Write tests");
        found.ifPresent(t -> System.out.println("Found: " + t.summary()));
    }
}

File: demo-vault/code/hello.rs
/// A sample Rust module demonstrating ownership, traits, and error handling.

use std::collections::HashMap;

#[derive(Debug, Clone, PartialEq)]
pub enum Priority {
    Low,
    Medium,
    High,
}

#[derive(Debug, Clone)]
pub struct Task {
    pub title: String,
    pub priority: Priority,
    pub done: bool,
}

impl Task {
    pub fn new(title: impl Into<String>, priority: Priority) -> Self {
        Self { title: title.into(), priority, done: false }
    }

    pub fn mark_done(&mut self) {
        self.done = true;
    }

    pub fn summary(&self) -> String {
        let status = if self.done { "x" } else { " " };
        format!("[{}] {} ({:?})", status, self.title, self.priority)
    }
}

pub struct TaskManager {
    tasks: Vec<Task>,
}

impl TaskManager {
    pub fn new() -> Self {
        Self { tasks: Vec::new() }
    }

    pub fn add(&mut self, task: Task) {
        self.tasks.push(task);
    }

    pub fn pending(&self) -> Vec<&Task> {
        self.tasks.iter().filter(|t| !t.done).collect()
    }

    pub fn count_by_priority(&self) -> HashMap<&Priority, usize> {
        let mut counts = HashMap::new();
        for task in &self.tasks {
            *counts.entry(&task.priority).or_insert(0) += 1;
        }
        counts
    }

    pub fn find_mut(&mut self, title: &str) -> Option<&mut Task> {
        self.tasks.iter_mut().find(|t| t.title == title)
    }
}

impl Default for TaskManager {
    fn default() -> Self {
        Self::new()
    }
}

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut manager = TaskManager::new();
    manager.add(Task::new("Learn Rust traits", Priority::High));
    manager.add(Task::new("Read the book", Priority::Low));
    manager.add(Task::new("Fix lifetime issue", Priority::High));

    println!("Pending tasks:");
    for task in manager.pending() {
        println!("  {}", task.summary());
    }

    println!("\nBy priority: {:?}", manager.count_by_priority());

    // Mutate a task
    if let Some(t) = manager.find_mut("Learn Rust traits") {
        t.mark_done();
        println!("Completed: {}", t.summary());
    }

    Ok(())
}
